backend/fms_core/viewsets/full_sample.py

```diff
@@ -0,0 +1,184 @@
+from collections import Counter
+from django.core import serializers
+import json
+
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from django.db.models import Count, Q, F
+from django.core.exceptions import ValidationError
+
+from datetime import datetime
+from ..utils import RE_SEPARATOR, float_to_decimal
+
+
+from fms_core.models import FullSample, Biosample, Sample, DerivedSample, SampleKind, Container, Individual, DerivedBySample
+from fms_core.serializers import FullSampleSerializer, FullSampleExportSerializer, SampleSerializer
+from fms_core.template_importer.importers import SampleSubmissionImporter, SampleUpdateImporter
+
+from fms_core.template_paths import SAMPLE_SUBMISSION_TEMPLATE, SAMPLE_UPDATE_TEMPLATE
+
+from ._constants import _full_sample_filterset_fields
+from ._utils import TemplateActionsMixin, _list_keys, versions_detail
+from fms_core.services.sample import create_full_sample
+
+class FullSampleViewSet(viewsets.ModelViewSet, TemplateActionsMixin):
+    queryset = FullSample.objects.select_related("individual", "container", "sample_kind", "biosample", "derived_sample").all()
+    serializer_class = FullSampleSerializer
+
+    ordering_fields = (
+        *_list_keys(_full_sample_filterset_fields),
+    )
+
+    filterset_fields = {
+        **_full_sample_filterset_fields,
+    }
+
+    template_action_list = [
+        {
+            "name": "Add Samples",
+            "description": "Upload the provided template with up to 384 new samples.",
+            "template": SAMPLE_SUBMISSION_TEMPLATE,
+            "importer": SampleSubmissionImporter,
+        },
+        {
+            "name": "Update Samples",
+            "description": "Upload the provided template with up to 384 samples to update.",
+            "template": SAMPLE_UPDATE_TEMPLATE,
+            "importer": SampleUpdateImporter,
+        }
+    ]
+
+    def get_queryset(self):
+        project_name_filter = self.request.query_params.get('sample__projects__name__icontains')
```

Comment:
This is a strange place to hide a filter ... Can't you make an annotation for this to list it as a regular filter ?